### ISSUE TITLE
Be more strict and prevent incomplete patterns

### ISSUE THREAD

**Issue by amitaibu** [digitallyinduced/ihp#1681]
Currently, we have the `-Wincomplete-patterns` flag, which results in easily overlooked warnings.
Can we convert this to a compiler error instead?

**Comment by amitaibu** [comment]
In the framework code, a couple of examples:

![image](https://github.com/digitallyinduced/ihp/assets/125707/eb3fe813-3e76-4222-9b50-127e12e9d68f)

![image](https://github.com/digitallyinduced/ihp/assets/125707/35c44c4c-d93e-46b1-a3bb-7e28f0a1e683)

**Comment by amitaibu** [comment]
But would be useful to have it on the app code as well

**Comment by mpscholten** [comment]
For the IHP code itself this then should be done together with https://github.com/digitallyinduced/ihp/issues/1640

**Comment by amitaibu** [comment]
Great.

Is it possible meanwhile to have an error just for the app code?

**Comment by mpscholten** [comment]
Yes 👍  should be possible. I'd keep this issue open actually, otherwise we might forget about this
